dotCMS/src/main/java/com/dotcms/contenttype/util/KeyValueFieldUtil.java

```diff
@@ -28,8 +28,16 @@
 				return keyValueMap;
 			}
 
+			// the following code fixes issue 10529
+			String json2;

```

Comment:
I think we should encapsulate this cleaning logic in a util or so, Also not sure why are not using the MarshalUtil with is the encapsulation of the gson